codesign/lib/src/file_codesign_visitor.dart

```diff
@@ -310,14 +315,27 @@ update these file paths accordingly.
         entitlementsFile.absolute.path
       ],
     ];
-    final io.ProcessResult result = await processManager.run(args);
-    if (result.exitCode != 0) {
-      throw CodesignException(
+
+    io.ProcessResult? result;
+    while (retryCount > 0) {
+      result = await processManager.run(args);
+      if (result.exitCode == 0) {
+        return;
+      }
+
+      log.severe(
         'Failed to codesign ${binaryFile.absolute.path} with args: ${args.join(' ')}\n'
         'stdout:\n${(result.stdout as String).trim()}'
         'stderr:\n${(result.stderr as String).trim()}',
       );
+
+      retryCount -= 1;
+      io.sleep(Duration(seconds: sleepTime));
```

Comment:
```suggestion
      await Future.delayed(Duration(seconds: sleepTime));
```

Reply:
Also, maybe we should double sleepTime each time we retry?

Reply:
Ah TIL about Future.delayed. and sure, doubled sleep time